experimental/table/server/master/src/main/java/alluxio/master/catalog/AlluxioCatalog.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.catalog;
+
+import alluxio.client.file.FileSystem;
+import alluxio.conf.PropertyKey;
+import alluxio.conf.ServerConfiguration;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import org.apache.hadoop.hive.metastore.api.Database;
+import org.apache.iceberg.BaseMetastoreCatalog;
+import org.apache.iceberg.BaseTable;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.TableOperations;
+import org.apache.iceberg.Tables;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+// TODO: (yuzhu) journal the state of the catalog
+public class AlluxioCatalog extends BaseMetastoreCatalog implements Closeable, Tables {
```

Comment:
I feel like there is a level of abstraction that is missing. The catalog should consist of databases, and each database should consist of tables. I think those should be abstractions/interfaces. Then, for a particular table, it is implemented as an iceberg table. That way, there is a clean separation between the two layers.

Reply:
i wanted to extend `BaseMetastoreCatalog` because I wanted to reuse some of the iceberg code there, and everything it returns is an iceberg table. 
I will do a major round of refactoring to reduce our dependency on iceberg table after this initial commit.